.github/workflows/python-versions.yml

```diff
@@ -0,0 +1,60 @@
+name: python versions
+on:
+  pull_request:
+    paths:
+    - 'requirements*'
+    - '.github/workflows/python-versions.yml'
+
+jobs:
+  # This workflow makes sure that Python dependencies install correctly for
+  # a) our current version b) the next version we're targetting and
+  # optionally c) a team request
+  python-versions:
+    name: set up
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 15
+    strategy:
+      matrix:
+        # macosx-10.15 is Catalina
+        # macosx-11.0 is Big Sur, however, it takes long for jobs to get started
+        os: [ macos-10.15, macos-11.0, ubuntu-20.04 ]
```

Comment:
I've added Big Sur last moment which might yield some errors.
It's far more important for me to get your initial feedback.